### PR TITLE
Find Application by Open Intern Position and Student

### PR DESCRIPTION
**Pull request by LarisaBothaza**

### CODE REVIEW

File: src/main/java/com/sintern/api/ApplicationController.java [hunk 1]
```diff
@@ -63,4 +66,14 @@ public void addApplication(@RequestBody AddApplicationRequest addApplicationRequ
 
         return applicationByInternPositionResponseList;
     }
+
+    @PostMapping("/applied")
+    @ResponseStatus(HttpStatus.OK)
+    public GetApplicationByOpenInternPositionAndStudentResponse findApplicationByOpenInternPositionIdAndStudentId(@RequestBody ApplicationByOpenInternPositionAndStudentRequest applicationByOpenInternPositionAnsStudentRequest) {
```
Comment (burticpatricia) [inline src/main/java/com/sintern/api/ApplicationController.java]: I think there is a typo in applicationByOpenInternPositionAnsStudentRequest :D (_AnsStudent_)

File: src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        this.studentID = studentID;
+    }
+
+    public UUID getOpenInternPositionID() {
+        return openInternPositionID;
+    }
+
+    public void setOpenInternPositionID(UUID openInternPositionID) {
+        this.openInternPositionID = openInternPositionID;
+    }
```
Comment (burticpatricia) [inline src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java]: Why don't we use Lombok annotation @ Getter and @ Setter, for cleaner code?

File: src/main/java/com/sintern/api/response/GetApplicationByOpenInternPositionAndStudentResponse.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    private UUID applicationID;
+
+    public UUID getApplicationID() {
+        return applicationID;
+    }
+
+    public void setApplicationID(UUID applicationID) {
+        this.applicationID = applicationID;
+    }
```
Comment (burticpatricia) [inline src/main/java/com/sintern/api/response/GetApplicationByOpenInternPositionAndStudentResponse.java]: same here

File: src/main/java/com/sintern/service/ApplicationServiceImpl.java [hunk 4]
```diff
@@ -67,4 +67,9 @@ public void addApplication(UUID studentID, UUID openInternPositionID, String des
     public List<Application> findApplicationsByOpenInternPosition(OpenInternPosition openInternPositionID) {
         return applicationRepository.findApplicationByOpenInternPosition(openInternPositionID);
     }
+
+    @Override
+    public Application findApplicationsByOpenInternPositionIdAndStudentId(UUID openInternPositionID, UUID studentID) {
+        return applicationRepository.findApplicationByStudentIdAndOpenInternPositionId(studentID, openInternPositionID);
+    }
 }
```
Comment (burticpatricia) [inline src/main/java/com/sintern/service/ApplicationServiceImpl.java]: please leave an empty line at the end of the file :)

**Review by burticpatricia — CHANGES_REQUESTED** [review]
please resolve this comments.
Good job! :D

File: src/main/java/com/sintern/api/ApplicationController.java (hunk 1 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/api/ApplicationController.java]: Ups, thank you

File: src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java (hunk 2 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java]: Ok, I'm making the change. But all response and request classes from API are like that. So they should be changed? @DascaluCosmin

File: src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.sintern.api.request;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.UUID;
+
+@Getter
+@Setter
```
Comment (burticpatricia) [inline src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java:9]: Nice :)

**Review by burticpatricia — APPROVED** [review]
Nice work! :)

File: src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java (hunk 2 above)
Comment (DascaluCosmin) [reply src/main/java/com/sintern/api/request/ApplicationByOpenInternPositionAndStudentRequest.java]: It's fine to keep them as they are.

**Review by DascaluCosmin — APPROVED** [review]

**Pull request merged**